sdk/metric/sdk.go

```diff
@@ -633,9 +640,19 @@ func (m *SDK) Resource() resource.Resource {
 }
 
 // RecordBatch enters a batch of metric events.
-func (m *SDK) RecordBatch(ctx context.Context, ls api.LabelSet, measurements ...api.Measurement) {
+func (m *SDK) RecordBatch(ctx context.Context, kvs []core.KeyValue, measurements ...api.Measurement) {
+	var labels labels
 	for _, meas := range measurements {
-		meas.SyncImpl().RecordOne(ctx, meas.Number(), ls)
+		s := meas.SyncImpl().(*syncInstrument)
+
+		h := s.acquireHandle(kvs, &labels)
```

Comment:
It seems like a large amount of the changes to this file are intended to optimize this loop, can we add some comments making that clear for future authors?

Reply:
Done.